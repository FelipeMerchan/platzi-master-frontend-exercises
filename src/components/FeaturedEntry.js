import React from 'react'

import featuredEntryImage from '../images/featured-entry.jpg'
import './styles/FeaturedEntry.css'

function FeaturedEntry() {
  return (
    <article className="FeaturedEntry mt-2 mb-2">
      <div className="FeaturedEntry__details">
        <h5 className="mb">
          Entrada destacada
        </h5>
        <h2 className="mb">
          Crea un sitio web usando HTML5
        </h2>
        <p className="FeaturedEntry__text">
          Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed 
          diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam 
          erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation 
          ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat. 
          Duis autem vel eum iriure dolor in hendrerit in vulputate velit esse 
          molestie consequat.
        </p>
      </div>
      <figure className="FeaturedEntry__imageContainer">
        <img className="FeaturedEntry__image" src={ featuredEntryImage } alt="Imagen principal" />
      </figure>
    </article>
  )
}

export default FeaturedEntry