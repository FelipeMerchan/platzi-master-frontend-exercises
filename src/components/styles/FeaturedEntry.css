.FeaturedEntry {
  display: flex;
  justify-content: space-between;
  padding: var(--space-2);
  border-radius: var(--border-radius);
  font-family: var(--font-2);
  background-color: var(--surfaceColor);
}

.FeaturedEntry__details {
  width: 68%;
  margin-right: 2%;
}

.FeaturedEntry__details>h2 {
  color: #c84451;
}

.FeaturedEntry__text {
  line-height: 1.2;
}

.FeaturedEntry__imageContainer {
  width: 30%;
  padding: 5px;
  align-self: center;
  outline: 1px solid var(--backgroundDarkGray);
}

.FeaturedEntry__image {
  width: 100%;
}

@media screen and (max-width: 768px) {
  .FeaturedEntry {
    display: block;
  }

  .FeaturedEntry__details,
  .FeaturedEntry__imageContainer {
    width: auto;
  }

  .FeaturedEntry__text {
    margin-bottom: var(--space-2);
  }
}