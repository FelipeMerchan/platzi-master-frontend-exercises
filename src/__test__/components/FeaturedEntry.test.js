import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import FeaturedEntry from '../../components/FeaturedEntry'

describe('Probar el componente <FeaturedEntry />', () => {
  const featuredEntry = shallow(<FeaturedEntry />)

  test('Render del componente FeaturedEntry', () => {
    expect(featuredEntry.length).toEqual(1)
  })
  test('Render de los 2 hijos', () => {
    expect(featuredEntry.children()).toHaveLength(2);
  });
  test('Render del tÃ­tulo', () => {
    expect(featuredEntry.find('h2').text()).toEqual('Crea un sitio web usando HTML5')
  })
  test('Render con el className FeaturedEntry__image', () => {
    expect(featuredEntry.exists('.FeaturedEntry__image')).toBeTruthy()
  })
})

describe('FeaturedEntry Snapshot', () => {
  test('Comprobar la UI del componente FeaturedEntry', () => {
    const featuredEntry = create(<FeaturedEntry />)
    expect(featuredEntry.toJSON()).toMatchSnapshot()
  })
})