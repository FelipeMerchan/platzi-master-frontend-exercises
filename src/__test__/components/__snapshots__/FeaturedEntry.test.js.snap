// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedEntry Snapshot Comprobar la UI del componente FeaturedEntry 1`] = `
<article
  className="FeaturedEntry mt-2 mb-2"
>
  <div
    className="FeaturedEntry__details"
  >
    <h5
      className="mb"
    >
      Entrada destacada
    </h5>
    <h2
      className="mb"
    >
      Crea un sitio web usando HTML5
    </h2>
    <p
      className="FeaturedEntry__text"
    >
      Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat. Duis autem vel eum iriure dolor in hendrerit in vulputate velit esse molestie consequat.
    </p>
  </div>
  <figure
    className="FeaturedEntry__imageContainer"
  >
    <img
      alt="Imagen principal"
      className="FeaturedEntry__image"
      src="featured-entry.jpg"
    />
  </figure>
</article>
`;
